add backend is available

cccclai opened the pull request:
Summary: Add a pybind API to see if the backend is available

Differential Revision: D69810445

mergennachin commented on extension/pybindings/pybindings.cpp:
```diff
@@ -990,6 +992,20 @@ py::list get_registered_backend_names() {
 
+py::bool_ is_available(const std::string& target_backend_name) {
+  size_t n_of_registered_backends = get_num_registered_backends();
+  for (size_t i = 0; i < n_of_registered_backends; i++) {
+    auto backend_name_res = get_backend_name(i);
+    THROW_IF_ERROR(backend_name_res.error(), "Failed to get backend name");
+    auto backend_name = backend_name_res.get();
+    if(strcmp(backend_name, target_backend_name.c_str()) == 0) {
+      BackendInterface* backend = get_backend_class(backend_name);
+      return backend->is_available();
+    }
+  }
```
No need to iterate here, just call `get_backend_class(target_backend_end)`, which does the iteration.

cccclai replied:
Ah yeah, updated.

cccclai committed:
add backend is available (#8738)

Summary:

Add a pybind API to see if the backend is available

Reviewed By: mergennachin

Differential Revision: D69810445
